app/controllers/application_controller.rb

```diff
@@ -3,4 +3,10 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :null_session
+
+  rescue_from ActiveRecord::RecordInvalid, with: :unprocessable_entity
+
+  def unprocessable_entity
+    render json: { error: "record invalid" }, status: 422
+  end
```

Comment:
Are you sure you need this? I wonder if rails catches this for us automatically - though even if it does I wonder if the route needs to be told it's a json format.

Reply:
I hope this is useful as it took me forever to look up (as it does every single time I try to look it up): if you check out: config.action_dispatch.rescue_responses on http://guides.rubyonrails.org/configuring.html and this section is interesting too: config.debug_exception_response_format

Reply:
Good spot, Rails does automatically catch `RecordInvalid` and respond 422, according to `rescue_responses` in that ^ documentation. I've removed the handling from this PR.

`debug_exception_response_format` on changes the format in development mode, which probably isn't what we want. Maybe we can just leave the default response alone for now as nothing's actually using it.